### PR TITLE
Refactoring of igv tracks building

### LINKED ISSUE #3139: Refactor the construction of IGV tracks
So that the things passed from variant page is institute, case and variant, instead of:

individuals and each individuals tracks (bam and bai), case groups and relative individual tracks (bam and bai), chromosome, start, end. All these things are now passed as args to the IGV endpoint.

This way the code will be much cleaner and organized. While refactoring add a check for user-specific cases in that code.

### LINKED ISSUE #3250: Found a possible security concern
Hey there!

I belong to an open source security research community, and a member (@porcupineyhairs) has found an issue, but doesn’t know the best way to disclose it.

If not a hassle, might you kindly add a `SECURITY.md` file with an email, or another contact method? GitHub [recommends](https://docs.github.com/en/code-security/getting-started/adding-a-security-policy-to-your-repository) this best practice to ensure security issues are responsibly disclosed, and it would serve as a simple instruction for security researchers in the future.

Thank you for your consideration, and I look forward to hearing from you!

(cc @huntr-helper)

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #3139 and fix #3250

- Build case IGV tracks starting from case and variant objects, and not passing all params from an HTML form
- Check that user has access to a case track before redirecting to IGV view (using [institute_and_case](https://github.com/Clinical-Genomics/scout/blob/4c177c838639cb648bbe03c4be6f710f3dcc4d2c/scout/server/utils.py#L89) function to retrieve case info and check if user is allowed to see resources)
- Check that user has access to RNA splice junctions before redirecting to IGV sashimi plot (also using institute_and_case function)

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
</details>

**How to test**:
1. Install on stage (cg-vm1) and open alignments:
- [ ] At the case level (from case sidebar)
- [ ] For a SNV variant
- [ ] For a SV variant
- [ ] For STRs
- [ ] For cancer variants
- [ ] For a case in a group with other cases

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by dnil** [comment]
Mental reminder: the current version seemed to have an issue with the cancer pages on the demo! 😊

**Comment by northwestwitch** [comment]
I think I'm ready for a review with this one!

**Comment by northwestwitch** [comment]
I was thinking that we already have enough bug fixes to create a new release soon. This PR is not an easy and fast review and we don't have to include it in the new release!

**Comment by dnil** [comment]
> I was thinking that we already have enough bug fixes to create a new release soon. This PR is not an easy and fast review and we don't have to include it in the new release!

Yeah, agree on the prio; we have a few immediate bug fixes and good features in main. It would be nice to have this fix out as well though. I should have some more dev time tomorrow again; let's see what there is time for!

File: scout/server/blueprints/alignviewers/views.py [hunk 1]
```diff
@@ -76,51 +80,44 @@ def unindexed_remote_static():
...
-        chromosome_build = "38"
-    else:
-        chromosome_build = "37"
-
-    start = request.form.get("start")
-    stop = request.form.get("stop")
-    locus = "chr{0}:{1}-{2}".format(chrom, start, stop)
-
-    display_obj = {}  # Initialize the dictionary containing all tracks info
+    _, case_obj = institute_and_case(
```
Comment (northwestwitch) [inline scout/server/blueprints/alignviewers/views.py:89]: Passing through this function makes the code safe (checks that institute_and_case function takes care of checking if the user is allowed to see the resource. There is no additional check required. Check the code executed in this function: https://github.com/Clinical-Genomics/scout/blob/4c177c838639cb648bbe03c4be6f710f3dcc4d2c/scout/server/utils.py#L89

I've also written a test at this regard (if your user is not authorized then you get a 403 (unathorized error message)

**Comment by northwestwitch** [comment]
I've been checking all other places in the code where  flask `send_file` is used to serve resources. There is nowhere else where the path to the resource is built based on params that can edited in a request by a user. Additionally, in all cases outside alignviewers the code either passes through `institute_and_case` function to collect case data (cases views) or creates the report based on the user object and its institutes (https://github.com/Clinical-Genomics/scout/blob/5ccf6b4d6cdc4f8dd27933f2b9b9503ae3d253ce/scout/server/blueprints/variants/views.py#L542). Therefore I think that when we merge this PR we can close the security problem issue!

**Comment by northwestwitch** [comment]
> Mental reminder: the current version seemed to have an issue with the cancer pages on stage!

This should be fixed now!

File: scout/server/blueprints/variant/templates/variant/buttons.html [hunk 2]
```diff
@@ -119,77 +119,3 @@
     </div>
   </form>
 {% endmacro %}
-
-{% macro igv_button(case, variant, case_groups) %}
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/buttons.html:123]: 🥇

**Review by dnil — APPROVED** [review]
Very nice! I have not tested through the details, but code looks good, and solves the issues at hand. There are automated tests, and local tests all check out nicely, but we should perhaps do a little more in depth stage testing with case groups before we deploy it.

**Pull request merged**